src/node.cc

```diff
@@ -2479,26 +2483,58 @@ struct node_module* get_linked_module(const char* name) {
   return mp;
 }
 
-// DLOpen is process.dlopen(module, filename).
+// DLOpen is process.dlopen(module, filename, flags).
 // Used to load 'module.node' dynamically shared objects.
 //
 // FIXME(bnoordhuis) Not multi-context ready. TBD how to resolve the conflict
 // when two contexts try to load the same shared object. Maybe have a shadow
 // cache that's a plain C list or hash table that's shared across contexts?
 static void DLOpen(const FunctionCallbackInfo<Value>& args) {
   Environment* env = Environment::GetCurrent(args);
+  bool is_dlopen_error = false;
+  const char *dlerrmsg = nullptr;
   uv_lib_t lib;
 
   CHECK_EQ(modpending, nullptr);
 
-  if (args.Length() != 2) {
-    env->ThrowError("process.dlopen takes exactly 2 arguments.");
+  if (args.Length() < 2) {
+    env->ThrowError("process.dlopen needs at least 2 arguments.");
     return;
   }
 
   Local<Object> module = args[0]->ToObject(env->isolate());  // Cast
   node::Utf8Value filename(env->isolate(), args[1]);  // Cast
-  const bool is_dlopen_error = uv_dlopen(*filename, &lib);
+
+#ifdef __POSIX__
+  int flags = 0;
+  if (args.Length() > 2) {
+    if (!args[2]->IsInt32() && !args[2]->IsUndefined()) {
+      return env->ThrowTypeError("dlopen flags must be undefined or integer.");
+    }
+    if (args[2]->IsInt32())
+      flags = args[2]->Int32Value();
```

Comment:
@bnoordhuis Would something like this (pattern taken from `node_crypto.cc`) work ?
```c++
  Maybe<int32_t> maybe_flags;
  if (args.Length() > 2) {
    CHECK(args[2]->IsInt32());
    maybe_flags = args[2]->Int32Value(env->context());
    CHECK(maybe_flags.IsJust());
  }
  // and later ....
  lib.handle = dlopen(*filename, maybe_flags.FromMaybe(RTLD_LAZY));
```

BTW, Isn't the first `CHECK(args[2]->IsInt32());` redundant since we are using a Maybe type?